eth/downloader/receiptreference.go

```diff
@@ -0,0 +1,125 @@
+package downloader
+
+import (
+	"bytes"
+	"embed"
+	"encoding/gob"
+	"path"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// userDepositNonces is a struct to hold the reference data for user deposits
+// The reference data is used to correct the deposit nonce in the receipts
+type userDepositNonces struct {
+	ChainID uint64 `json:"chainId"`
+	// "Start" and "End" would be more appropriate names. Last is not inclusive.
+	First   uint64              `json:"start"`
+	Last    uint64              `json:"end"`
+	Results map[uint64][]uint64 `json:"results"`
+}
+
+var userDepositNoncesReference = make(map[uint64]userDepositNonces)
+
+//go:embed userDepositData/*.gob
+var receiptReference embed.FS
+
+var receiptReferencePath = "userDepositData"
+
+// lazy load all user deposit nonce reference data for all chains in the userDepositData directory
+func initReceiptReferences() {
```

Comment:
Can we do this per network to not have to load every single map?

Reply:
Sure! I had originally avoided that because I wasn't assuming a well formatted filename, but now that they're all here, it's easy to inspect the chainID